lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/persistence/DBOChange.java

```diff
@@ -161,110 +145,139 @@ public void setParentId(Long parentId) {
+			if (other.changeNumber != null)
 				return false;
-			}
-		} else if (!changeNumber.equals(other.changeNumber)) {
+		} else if (!changeNumber.equals(other.changeNumber))
 			return false;
-		}
-		if (changeType != other.changeType) {
+		if (changeType != other.changeType)
 			return false;
-		}
 		if (objectEtag == null) {
-			if (other.objectEtag != null) {
+			if (other.objectEtag != null)
 				return false;
-			}
-		} else if (!objectEtag.equals(other.objectEtag)) {
+		} else if (!objectEtag.equals(other.objectEtag))
 			return false;
-		}
 		if (objectId == null) {
-			if (other.objectId != null) {
+			if (other.objectId != null)
 				return false;
-			}
-		} else if (!objectId.equals(other.objectId)) {
+		} else if (!objectId.equals(other.objectId))
 			return false;
-		}
-		if (objectType != other.objectType) {
+		if (objectType != other.objectType)
 			return false;
-		}
 		if (parentId == null) {
-			if (other.parentId != null) {
+			if (other.parentId != null)
 				return false;
-			}
-		} else if (!parentId.equals(other.parentId)) {
+		} else if (!parentId.equals(other.parentId))
 			return false;
-		}
 		if (timeStamp == null) {
-			if (other.timeStamp != null) {
+			if (other.timeStamp != null)
 				return false;
-			}
-		} else if (!timeStamp.equals(other.timeStamp)) {
+		} else if (!timeStamp.equals(other.timeStamp))
+			return false;
+		if (userId == null) {
+			if (other.userId != null)
+				return false;
+		} else if (!userId.equals(other.userId))
 			return false;
-		}
 		return true;
 	}
 
-
-	/* (non-Javadoc)
-	 * @see java.lang.Object#toString()
-	 */
 	@Override
 	public String toString() {
-		return "DBOChange [changeNumber=" + changeNumber + ", timeStamp="
-				+ timeStamp + ", objectId=" + objectId + ", parentId="
-				+ parentId + ", objectType=" + objectType + ", objectEtag="
-				+ objectEtag + ", changeType=" + changeType + "]";
+		return "DBOChange [changeNumber=" + changeNumber + ", timeStamp=" + timeStamp + ", objectId=" + objectId
+				+ ", parentId=" + parentId + ", objectType=" + objectType + ", objectEtag=" + objectEtag
+				+ ", changeType=" + changeType + ", userId=" + userId + "]";
 	}
 
+	@Override
+	public TableMapping<DBOChange> getTableMapping() {
+		return new TableMapping<DBOChange>(){
+
+			@Override
+			public DBOChange mapRow(ResultSet rs, int rowNum) throws SQLException {
+				DBOChange dbo = new DBOChange();
+				dbo.setChangeNumber(rs.getLong(COL_CHANGES_CHANGE_NUM));
+				dbo.setTimeStamp(rs.getTimestamp(COL_CHANGES_TIME_STAMP));
+				dbo.setObjectId(rs.getLong(COL_CHANGES_OBJECT_ID));
+				dbo.setObjectType(rs.getString(COL_CHANGES_OBJECT_TYPE));
+				Long userId = rs.getLong(COL_CHANGES_USER_ID);
+				if (!rs.wasNull()) {
+					dbo.setUserId(userId);
+				}
+				Long parentId = rs.getLong(COL_CHANGES_PARENT_ID);
+				if (!rs.wasNull()) {
+					dbo.setParentId(parentId);
+				}
+				String etag = rs.getString(COL_CHANGES_OBJECT_ETAG);
+				if (!rs.wasNull()) {

```

Comment:
was null not needed for get() that returns an object.  Only needed for get() return primitive.